app/controllers/interviews_controller.rb

```diff
@@ -69,6 +85,29 @@ def set_interview
 
     # Only allow a list of trusted parameters through.
     def interview_params
-      params.require(:interview).permit(:title, :participants_email, :start_time, :end_time)
+      params.require(:interview).permit(:title, :start_time, :end_time, :interviewer_id, :interviewee_id) 
+    end
+
+    def participants_available
+      logger.info((params[:interview]["title"]))
+      start = DateTime.new(params[:interview]["start_time(1i)"].to_i, 
+      params[:interview]["start_time(2i)"].to_i,
+      params[:interview]["start_time(3i)"].to_i,
+      params[:interview]["start_time(4i)"].to_i,
+      params[:interview]["start_time(5i)"].to_i)
+      close = DateTime.new(params[:interview]["end_time(1i)"].to_i, 
+      params[:interview]["end_time(2i)"].to_i,
+      params[:interview]["end_time(3i)"].to_i,
+      params[:interview]["end_time(4i)"].to_i,
+      params[:interview]["end_time(5i)"].to_i)
+      # logger.info("here" + params[:interview][:interviewer_id] + params[:interview]["start_time(1i)"])#.strftime('%a %b %d %H:%M:%S %Z %Y') )# + params[:interview][:end_time])
+      time_clash = Interview.where("(end_time BETWEEN ? AND ?) or (start_time BETWEEN ? AND ?) or ((end_time > ?) and (start_time < ?))", start, close, start, close, close, start)
+      if time_clash.where("interviewer_id = ?", params[:interview][:interviewer_id]).exists?
+        return 1
+      end
+      if time_clash.where("interviewee_id = ?", params[:interview][:interviewee_id]).exists?
+        return 2
+      end
+      return 0
```

Comment:
u r first checking all the interviews that clashes and thn checking ki if the clashed interviews is of particular participant ... Suppose if the number of interviews is really really high at a particular time say(4pm-6pm) thn dont u think this approach will take a little longer. Can u come up with better approach ???